ludwig/config_validation/checks.py

```diff
@@ -631,6 +631,25 @@ def check_llm_quantization_backend_incompatibility(config: "ModelConfig") -> Non
             pass
 
 
+@register_config_check
+def check_llm_text_encoder_is_not_used_with_ecd(config: "ModelConfig") -> None:
+    """Checks that a pretrained text encoder is not used for ECD models with a text output feature."""
+    if config.model_type != MODEL_ECD:
+        return
+
+    if config.output_features[0].type != TEXT:
```

Comment:
@justinxzhao Very minor -- is there a reason to specify "output" first, instead of "input" (cognitive strain)?  Thanks.

Reply:
Done.